### ISSUE TITLE
dev branch - model definition notes

### ISSUE THREAD

**Issue by dsyme** [DiffSharp/DiffSharp#66]
The model definition in `Tests.fs` could be like this:

```fsharp

let MakeModel ps f =
    let model = 
        { new  Model() with
            override l.Forward(x) = f x 
          }
    model.AddParameters(ps)
    model

let FeedforwardNet(p2: int) =
    let fc1 = Linear(2, p2)
    let fc2 = Linear(p2, 1)
    MakeModel ["fc1", fc1; "fc2", fc2] 
        (fc1.Forward >> 
         Tensor.LeakyRelu >> 
         fc2.Forward >> 
         Tensor.LeakyRelu)

let Compose (model1: Model) (model2: Model) =
    MakeModel [ for KeyValue(k,v) in model1.Parameters do yield (k,v)
                for KeyValue(k,v) in model2.Parameters do yield (k,v) ]
        (model1.Forward >> model2.Forward)
        
```

**Comment by MikaelUmaN** [comment]
Looks great. Does this mean diffsharp will grow into something similar to pytorch. If so, that’s awesome.

**Comment by gbaydin** [comment]
Closing this as the API proposed by @dsyme is implemented and merged in #96. An example is here https://github.com/DiffSharp/DiffSharp/blob/4dc11c7a36ef12dfd5594eb5c45bdf4e9111e73f/src/DiffSharp.Tests/TestModel.fs#L74
